packages/tasks/src/tasks/text-generation/inference.ts

```diff
@@ -5,204 +5,94 @@
  */
 
 /**
- * Inputs for Text Generation inference
+ * Text Generation Input
  */
 export interface TextGenerationInput {
-	/**
-	 * The text to initialize generation with
-	 */
 	inputs: string;
-	/**
-	 * Additional inference parameters
-	 */
-	parameters?: TextGenerationParameters;
-	/**
-	 * Whether to stream output tokens
-	 */
-	stream?: boolean;
+	parameters?: TextGenerationInputGenerateParameters;
 	[property: string]: unknown;
 }
 
-/**
- * Additional inference parameters
- *
- * Additional inference parameters for Text Generation
- */
-export interface TextGenerationParameters {
-	/**
-	 * The number of sampling queries to run. Only the best one (in terms of total logprob) will
-	 * be returned.
-	 */
+export interface TextGenerationInputGenerateParameters {
 	best_of?: number;
-	/**
-	 * Whether or not to output decoder input details
-	 */
 	decoder_input_details?: boolean;
-	/**
-	 * Whether or not to output details
-	 */
 	details?: boolean;
-	/**
-	 * Whether to use logits sampling instead of greedy decoding when generating new tokens.
-	 */
 	do_sample?: boolean;
-	/**
-	 * The maximum number of tokens to generate.
-	 */
+	frequency_penalty?: number;
+	grammar?: TextGenerationInputGrammarType;
 	max_new_tokens?: number;
-	/**
-	 * The parameter for repetition penalty. A value of 1.0 means no penalty. See [this
-	 * paper](https://hf.co/papers/1909.05858) for more details.
-	 */
 	repetition_penalty?: number;
-	/**
-	 * Whether to prepend the prompt to the generated text.
-	 */
 	return_full_text?: boolean;
-	/**
-	 * The random sampling seed.
-	 */
 	seed?: number;
-	/**
-	 * Stop generating tokens if a member of `stop_sequences` is generated.
-	 */
-	stop_sequences?: string[];
-	/**
-	 * The value used to modulate the logits distribution.
-	 */
+	stop?: string[];
 	temperature?: number;
-	/**
-	 * The number of highest probability vocabulary tokens to keep for top-k-filtering.
-	 */
 	top_k?: number;
-	/**
-	 * If set to < 1, only the smallest set of most probable tokens with probabilities that add
-	 * up to `top_p` or higher are kept for generation.
-	 */
+	top_n_tokens?: number;
 	top_p?: number;
-	/**
-	 * Truncate input tokens to the given size.
-	 */
 	truncate?: number;
-	/**
-	 * Typical Decoding mass. See [Typical Decoding for Natural Language
-	 * Generation](https://hf.co/papers/2202.00666) for more information
-	 */
 	typical_p?: number;
+	watermark?: boolean;
+	[property: string]: unknown;
+}
+
+export interface TextGenerationInputGrammarType {
+	type: Type;
 	/**
-	 * Watermarking with [A Watermark for Large Language Models](https://hf.co/papers/2301.10226)
+	 * A string that represents a [JSON Schema](https://json-schema.org/).
+	 *
+	 * JSON Schema is a declarative language that allows to annotate JSON documents
+	 * with types and descriptions.
 	 */
-	watermark?: boolean;
+	value: unknown;
 	[property: string]: unknown;
 }
 
+export type Type = "json" | "regex";
```

Comment:
This new type's name is a bit too generic IMO

Reply:
Not sure what I can do about this unfortunately :confused:
It comes from

```js
"GrammarType": {
  "oneOf": [
    {
      "type": "object",
      "required": ["type", "value"],
      "properties": {
        "type": { "type": "string", "enum": ["json"] },
        "value": { "type": "string" }
      }
    },
    {
      "type": "object",
      "required": ["type", "value"],
      "properties": {
        "type": { "type": "string", "enum": ["regex"] },
        "value": { "type": "string" }
      }
    }
  ],
  "discriminator": { "propertyName": "type" }
}
``` 
in the original JSON specs. I tried playing with the import script but I feel a fix would have to live in `quicktype` itself (kinda annoying to fix/debug TBH)